packages/transport-tcp/test/connection-time.spec.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @packageDocumentation
+ * 
+ * Tests for TCP connection establishment time.
+ * This test verifies that the connection time is below an acceptable threshold.
+ */
+
+import { expect } from 'aegir/chai'
+import net from 'node:net'
+import { tcp } from '../src/index.js'
+import { createLibp2p } from 'libp2p'
+import { multiaddr } from '@multiformats/multiaddr'
+
+describe('TCP connection time', () => {
+  it('should establish a connection in under 300ms', async () => {
+    // Create a simple TCP server
+    const server = net.createServer()
+    const port = 8080 + Math.floor(Math.random() * 1000)
+    
+    await new Promise(resolve => {
+      server.listen(port, '127.0.0.1', resolve)
+    })
+    
+    try {
+      // Measure connection time
+      const start = Date.now()
+      
+      const socket = await new Promise((resolve, reject) => {
+        const socket = net.connect(port, '127.0.0.1')
+        socket.on('connect', () => resolve(socket))
+        socket.on('error', reject)
+      })
+      
+      const connectionTime = Date.now() - start
+      console.log(`TCP connection established in ${connectionTime}ms`)
+      
+      // Close the socket
+      socket.end()
+      
+      // Verify connection time is under threshold
+      expect(connectionTime).to.be.below(300, 'Connection time should be under 300ms')
```

Comment:
This only runs locally so yes, the connection time will be significantly under 300ms.